framework_tvb/tvb/core/adapters/inputs_processor.py

```diff
@@ -0,0 +1,94 @@
+            attr_default = getattr(form_model, field.name)
+            attr_vm = getattr(view_model, field.name)
+            if attr_vm != attr_default:
+                if isinstance(attr_default, float) or isinstance(attr_default, str):
+                    changed_attr[field.label] = attr_vm
+                else:
+                    changed_attr[field.label] = attr_vm.title
+        else:
+            attr_vm = getattr(view_model, field.name)
+            data_type = ABCAdapter.load_entity_by_gid(attr_vm)
+            if attr_vm:
+                changed_attr[field.label] = data_type.display_name
+            inputs_datatypes.append(data_type)
+
+    return inputs_datatypes, changed_attr
+
+
+def review_operation_inputs_from_adapter(adapter, operation):
+    """
+    :returns: a list with the inputs from the parameters list that are instances of DataType,\
+        and a dictionary with all parameters which are different than the declared defauts
+    """
+    view_model = adapter.load_view_model(operation)
+    form_model = adapter.get_view_model_class()()
+    form_fields = adapter.get_form_class()().fields
+
+    if 'SimulatorAdapter' in operation.algorithm.classname:
+        fragments = adapter.get_simulator_fragments()
+        inputs_datatypes, changed_attr = _review_operation_inputs_for_adapter_model(form_fields, form_model, view_model)
+        for fragment in fragments:
+            fragment_fields = fragment().fields
+            for field in fragment_fields:
+                if hasattr(view_model, field.name):
+                    if not isinstance(field, TraitDataTypeSelectField) and not isinstance(field, DataTypeSelectField):
+                        attr_default = getattr(form_model, field.name)
+                        attr_vm = getattr(view_model, field.name)
+                        if attr_vm != attr_default:
+                            if isinstance(attr_default, float) or isinstance(attr_default, str):
+                                changed_attr[field.label] = attr_vm
+                            else:
+                                if not isinstance(attr_default, tuple):
+                                    changed_attr[field.label] = attr_vm.title
+                                else:
+                                    for sub_attr in attr_default:
+                                        changed_attr[field.label] = sub_attr.title
+                    else:
+                        attr_vm = getattr(view_model, field.name)
+                        data_type = ABCAdapter.load_entity_by_gid(attr_vm)
+                        if attr_vm:
+                            changed_attr[field.label] = data_type.display_name
+                        inputs_datatypes.append(data_type)
+    else:
+        inputs_datatypes, changed_attr = _review_operation_inputs_for_adapter_model(form_fields, form_model, view_model)
+
```

Comment:
Maybe it would be easier to read this method if the` if-else` statements were inversed (maybe no `else` needed if the return is done inside the `if`)